### PR TITLE
Terraform 0.9.0, Generative TF Files, and more

### PR DESCRIPTION
**Pull request by jacknagz**
to @austinbyers @strcrzy 

## Background
When HashiCorp upgraded Terraform to 0.9.0, they deprecated `remote_states` in favor of `backend` ([info here](https://www.terraform.io/upgrade-guides/0-9.html)).  This PR makes StreamAlert future proof by moving to `backend`, and includes boilerplate code to dynamically generate entire cluster files (more details below).

To get `backend` to work, we first Terraform several resources in the init phase with local state.  After, the `s3` backend is enabled and then the remainder infrastructure is built.  The same steps happen in reverse during destruction.

## Changes
* Remove `remote_state_refresh()`: This is a deprecated command in Terraform > 0.9.0.
* Remove Jinja templating in favor of generating JSON.
* Better reliability of Terraform commands: Break execution if preceding steps fail.
* Cleanup cluster files on destroy, along with leftover state files.
* Remove default example SNS topic: This breaks Terraform (since it's not a valid ARN)
* Suppress `pip` logger messages

## Upgrade Notes
For users upgrading from 0.8.0 to 0.9.0 state files, run the following commands:

```
With the older Terraform version (version 0.8.x), run terraform remote pull.
This will cache the latest legacy remote state data locally. 
We'll use this for a backup in case things go wrong.

Backup your terraform/.terraform/terraform.tfstate file. 
This contains the cache we just pulled. 
Please copy this file to a location outside of your Terraform module.

# Note: this will wipe any custom Terraform added (at this time)
$ ./stream_alert_cli.py terraform generate
$ ./stream_alert_cli.py terraform init-backend
# Type 'no' at the prompt since your statefile isn't moving

Verify your state looks good by running stream_alert_cli.py terraform build and 
seeing if it detects your infrastructure.
```

For any other issues with upgrading, see the link above in `Background`

### CODE REVIEW

File: stream_alert_cli/runner.py [hunk 1]
```diff
@@ -14,16 +14,20 @@
 limitations under the License.
 '''
 
+import os
+import sys
+import shutil
```
Comment (austinbyers) [inline stream_alert_cli/runner.py]: Nit: switch `sys` and `shutil` for alphabetical import order

File: stream_alert_cli/terraform_generate.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        tf_file.write(main_json)
+
+    # Break out early during the init process, clusters aren't needed yet
+    if init:
+        return True
+
+    # Setup clusters
+    for cluster in config['clusters'].keys():
+        if cluster == 'main':
+            raise InvalidClusterName('Rename cluster "main" to something else!')
```
Comment (austinbyers) [inline stream_alert_cli/terraform_generate.py]: It doesn't look like `InvalidClusterName` is defined in this file? It's defined in `runner.py`

Comment (jacknagz) [inline stream_alert_cli/terraform_generate.py]: good catch, #copypasta

**Comment by jacknagz** [comment]
Related to #44

**Pull request merged**